src/vi/hangtruyen/src/eu/kanade/tachiyomi/extension/vi/hangtruyen/HangTruyen.kt

```diff
@@ -0,0 +1,163 @@
+package eu.kanade.tachiyomi.extension.vi.hangtruyen
+
+import android.content.SharedPreferences
+import android.widget.Toast
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.multisrc.wpcomics.WPComics
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.util.asJsoup
+import keiyoushi.utils.getPreferences
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class HangTruyen :
+    WPComics(
+        "HangTruyen",
+        "https://hangtruyen.net",
+        "vi",
+        dateFormat = SimpleDateFormat("dd-MM-yyyy", Locale.ROOT).apply {
+            timeZone = TimeZone.getTimeZone("Asia/Ho_Chi_Minh")
+        },
+        gmtOffset = null,
+    ),
+    ConfigurableSource {
+
+    // Popular
+    override fun popularMangaRequest(page: Int) =
+        GET("$baseUrl/tim-kiem?r=newly-updated&page=$page&orderBy=view_desc")
+
+    override fun popularMangaSelector() = "div.search-result div.row"
+
+    override fun popularMangaNextPageSelector() = ".next-page"
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val entries = document.select("div.search-result .m-post").map(::popularMangaFromElement)
```

Comment:
use `popularMangaSelector`? and remove function override it as it calls `popularMangaSelector` by default